### PR TITLE
Feat: New rule: fastqc

### PR DESCRIPTION
**Pull request by jonca79**

### CODE REVIEW

**Comment by Smeds** [comment]
Are all those .cache and .java directories supposed to be included?

File: .tests/integration/config.yaml [hunk 1]
```diff
@@ -3,3 +3,6 @@ samples: "samples.tsv"
 units: "units.tsv"
 
 default_container: "docker://marrip/common:1.1.0"
+
+fastqc:
+  container: "docker://gmsuppsala/somatic:develop"
```
Comment (Smeds) [inline .tests/integration/config.yaml:8]:
```suggestion
  container: "docker://hydra/fastqc:0.11.9"
```

File: .tests/integration/config.yaml [hunk 2]
```diff
@@ -3,3 +3,6 @@ samples: "samples.tsv"
 units: "units.tsv"
 
 default_container: "docker://marrip/common:1.1.0"
```
Comment (Smeds) [inline .tests/integration/config.yaml:5]:
```suggestion
default_container: "docker://hydra/common:0.0.1"
```

File: config/config.yaml [hunk 3]
```diff
@@ -4,5 +4,6 @@
  samples: "samples.tsv"
  units: "units.tsv"
 
- tools:
-   common: "docker://hydra/common:1.1.0"
+ default_container: "docker://marrip/common:1.1.0"
```
Comment (Smeds) [inline config/config.yaml:7]:
```suggestion
 default_container: ""docker://hydra/common:0.0.1""
 
 fastqc:
  container: "docker://hydra/fastqc:0.11.9"
```

File: workflow/rules/fastqc.smk [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        repeat("qc/fastqc/{sample}_{type}_{read}.fastqc.benchmark.tsv", config.get("fastqc", {}).get("benchmark_repeats", 1))
+    threads: config.get("fastqc", config["default_resources"]).get("threads", config["default_resources"]["threads"])
+    container:
+        config.get("fastqc", {}).get("container", config["default_container"])
+    conda:
+        "../envs/fastqc.yaml"
+    message:
+        "{rule}: qc/{rule}/{wildcards.sample}_{wildcards.type}_{wildcards.read}"
+    shell:
+        "(touch {output.tmp} && "
```
Comment (Smeds) [inline workflow/rules/fastqc.smk:32]: why not use wrapper? 

Is tmp.txt file really necessary?

File: workflow/rules/fastqc.smk [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        zip=temp("qc/fastqc/{sample}_{type}_{read}_fastqc.zip"),
+        tmp=temp("qc/fastqc/{sample}_{type}_{read}/tmp.txt"),
+    params:
+        outdir=lambda wildcards, output: os.path.dirname(os.path.abspath(output.html)),
+        tmp=lambda wildcards, output: os.path.dirname(os.path.abspath(output.tmp)),
+    log:
+        "qc/fastqc/{sample}_{type}_{read}.fastqc.log",
+    benchmark:
+        repeat("qc/fastqc/{sample}_{type}_{read}.fastqc.benchmark.tsv", config.get("fastqc", {}).get("benchmark_repeats", 1))
+    threads: config.get("fastqc", config["default_resources"]).get("threads", config["default_resources"]["threads"])
```
Comment (Smeds) [inline workflow/rules/fastqc.smk:24]:
```suggestion
    threads: config.get("fastqc", {}).get("threads", config["default_resources"]["threads"])
    resources:
        threads=config.get("fastqc", {}).get("threads", config["default_resources"]["threads"]),
        time=config.get("fastqc", {}).get("time", config["default_resources"]["time"]),
```

**Review by Smeds — CHANGES_REQUESTED** [review]

File: workflow/rules/fastqc.smk (hunk 4 above)
Comment (jonca79) [reply workflow/rules/fastqc.smk:32]: The wrapper / version used in the wrapper did not work last time I tried.
I need to create the tmp directory somehow.

File: .tests/integration/config.yaml [hunk 6]
```diff
@@ -2,4 +2,7 @@ resources: "resources.yaml"
 samples: "samples.tsv"
 units: "units.tsv"
 
-default_container: "docker://marrip/common:1.1.0"
+default_container: "docker://hydra/common:0.0.1"
+
+fastqc:
+  container: "docker://hydra/fastqc:0.11.9"
```
Comment (Smeds) [inline .tests/integration/config.yaml:8]:
```suggestion
  container: "docker://hydragenetics/fastqc:0.11.9"
```

**Review by Smeds — APPROVED** [review]

**Pull request merged**